bin/api-docs/are-readmes-unstaged.js

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env node
+
+/**
+ * Node dependencies.
+ */
+const { join } = require( 'path' );
+const chalk = require( 'chalk' );
+const execSync = require( 'child_process' ).execSync;
+
+/**
+ * Local dependencies.
+ */
+const getPackages = require( './packages' );
+
+const getUnstagedFiles = () => execSync( 'git diff --name-only', { encoding: 'utf8' } ).split( '\n' ).filter( ( element ) => '' !== element );
+
+const readmeFiles = getPackages().map( ( [ packageName ] ) => join( 'packages', packageName, 'README.md' ) );
```

Comment:
If we aim to remove the whitelist in #18840, what should we do here instead?

Do we need to care about whether it's a whitelisted README, or would it be enough to just flag _any_ README ? Or, like in #18840, do some scan of the content to see if it includes any relevant tags.

Do you think the former naive check would ever be problematic? I can see where I might want to commit some README changes separately, but that could happen even with the logic as proposed here. The most durable solution I think would require that we check whether the specific changes in a README.md occur within the opening and closing tags.

Reply:
A few things here:

- We should check only the READMEs that contain a token, otherwise, we may see many false positives.
- If we agree that the way to go is to merge this PR and then merge #18840, I would keep the README whitelist in this PR. There are other things in that PR that are relevant here as well, such as using the lint-staged files.
- If there are changes to READMEs that are unrelated to API docs, they'll be flagged to be added in the commit. This is a false positive. Note that this can be bypassed manually. Also, for context, in the past few months (I checked as far as August), most of the changes that happened to the tracked READMEs were about the API docs.
- Although checking that the specific changes at README happen within tokens would be the _perfect_ solution, it is far more complex than this. This is a good improvement that we can always improve later if it proves necessary.